app/uk/gov/hmrc/helptosavefrontend/controllers/EligibilityCheckController.scala

```diff
@@ -50,57 +51,66 @@ class EligibilityCheckController  @Inject()(val messagesApi: MessagesApi,
+        checkIfAlreadyEnrolled({
           nino ⇒
-            checkSession(
+            checkSession{
               // there is no session yet
-              performEligibilityChecks(nino, htsContext).fold(
-                handleEligibilityCheckError,
-                r ⇒ handleEligibilityResult(r, nino)
-              ), session ⇒
-                // there is a session
-                session.eligibilityCheckResult.fold(
-                  SeeOther(routes.EligibilityCheckController.notEligible().url)
-                )( _ ⇒
-                  SeeOther(routes.EligibilityCheckController.getIsEligible().url)
-                )
-            )
+              getEligibilityPlayResult(nino)
+            } { session ⇒
+              // there is a session
+              session.eligibilityCheckResult.fold(
+                // user is not eligible
+                SeeOther(routes.EligibilityCheckController.notEligible().url)
+              )(_ ⇒
+                // user is eligible
+                SeeOther(routes.EligibilityCheckController.getIsEligible().url)
+              )
+            }
+        }, { enrolmentCheckError ⇒
+          // if there is an error checking the enrolment, do the eligibility checks
+          getEligibilityPlayResult(enrolmentCheckError.nino)
         }
+        )
   }
 
   val notEligible: Action[AnyContent] = authorisedForHtsWithInfo {
     implicit request ⇒
       implicit htsContext ⇒
         checkIfAlreadyEnrolled { _ ⇒
-          checkSession(
-            SeeOther(routes.EligibilityCheckController.getCheckEligibility().url),
+          checkSession{
+            SeeOther(routes.EligibilityCheckController.getCheckEligibility().url)} {
             _.eligibilityCheckResult.fold {
               Ok(views.html.core.not_eligible())
             }(_ ⇒
               SeeOther(routes.EligibilityCheckController.getIsEligible().url)
             )
-          )
+          }
         }
   }
 
   val getIsEligible: Action[AnyContent] = authorisedForHtsWithInfo {
     implicit request ⇒
       implicit htsContext ⇒
         checkIfAlreadyEnrolled { _ ⇒
-          checkSession(
-            SeeOther(routes.EligibilityCheckController.getCheckEligibility().url),
+          checkSession{
+            SeeOther(routes.EligibilityCheckController.getCheckEligibility().url)
+          } {
             _.eligibilityCheckResult.fold(
               SeeOther(routes.EligibilityCheckController.notEligible().url)
             )(_ ⇒
               Ok(views.html.register.you_are_eligible())
             )
-          )
+          }
         }
   }
 
-  private def performEligibilityChecks(nino: NINO,
-                                       htsContext: HtsContext
-                                      )(implicit hc: HeaderCarrier): EitherT[Future, Error, EligibilityResultWithUserInfo] =
+  private def getEligibilityPlayResult(nino: NINO)(implicit hc: HeaderCarrier, htsContext: HtsContext, request: Request[AnyContent]) =
```

Comment:
The word play is meaningless, maybe getEligibilityHTTPResult or getEligibilityActionResult??